### PR TITLE
Implement user-permission API on admin permissions & add user-permission model

### LINKED ISSUE #549: Implement user permissions API on admin/permissions/system-roles
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Implement user-permission API on admin permissions

#### Changes proposed in this pull request:

- Add user-permission model
- Implement user-permission API on admin permissions
- Remove verified user column from the table

Fixes #549

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
@fossasia/open-event-frontend Please review

**Comment by CosmicCoder96** [comment]
@harshitagupta30 http://cc96-opev-fe-delta.herokuapp.com/admin/permissions/system-roles

**Comment by CosmicCoder96** [comment]
@niranjan94 @geekyd Please review

**Comment by harshitagupta30** [comment]
@niranjan94 From where can I get admin rights to check? Whenever I am trying to save getting the error "An unexpected error has occured. User permissions not saved."

File: app/controllers/admin/permissions/system-roles.js [hunk 1]
```diff
@@ -8,6 +8,19 @@ export default Controller.extend({
...
+      this.set('isLoading', true);
+      let userPermissions = this.get('model');
+      userPermissions.save()
+        .then(() => {
+          this.set('isLoading', false);
+          this.notify.success(this.l10n.t('User permissions have been saved successfully.'));
+        })
+        .catch(()=> {
+          this.set('isLoading', false);
+          this.notify.error(this.l10n.t('An unexpected error has occured. User permissions not saved.'));
```
Comment (harshitagupta30) [inline app/controllers/admin/permissions/system-roles.js]: @CosmicCoder96 Can you please update the message of occured? The correct one is 'occurred'. :)

Comment (CosmicCoder96) [inline app/controllers/admin/permissions/system-roles.js]: @harshitagupta30 Done!

File: app/routes/admin/permissions/system-roles.js [hunk 2]
```diff
@@ -5,5 +5,8 @@ const { Route } = Ember;
 export default Route.extend({
   titleToken() {
     return this.l10n.t('System Roles');
+  },
+  model() {
+    return this.get('store').query('user-permission', {});
```
Comment (niranjan94) [inline app/routes/admin/permissions/system-roles.js]: why query ? Wouldn't `findAll` be best suited for this ?

**Comment by CosmicCoder96** [comment]
@niranjan94 Can you please review again, thank you!

File: app/routes/admin/permissions/system-roles.js [hunk 3]
```diff
@@ -5,5 +5,16 @@ const { Route } = Ember;
...
   titleToken() {
     return this.l10n.t('System Roles');
+  },
+  model() {
+    return this.get('store').findAll('user-permission');
+  },
+  actions: {
+    willTransition() {
+      let model = this.get('controller.model');
+      model.forEach(function(userPermission) {
```
Comment (niranjan94) [inline app/routes/admin/permissions/system-roles.js]: Why store it in a variable ? Do a foreach directly on `this.get('controller.model')`

File: app/controllers/admin/permissions/system-roles.js [hunk 4]
```diff
@@ -8,6 +8,19 @@ export default Controller.extend({
     },
     addSystemRole() {
       this.set('isAddSystemRoleModalOpen', false);
+    },
+    updatePermissions() {
+      this.set('isLoading', true);
+      let userPermissions = this.get('model');
```
Comment (niranjan94) [inline app/controllers/admin/permissions/system-roles.js]: Why store in variable ? Use save on `this.get('model')`

**Comment by CosmicCoder96** [comment]
@niranjan94 I have made those changes.

File: app/routes/admin/permissions/system-roles.js [hunk 5]
```diff
@@ -5,5 +5,15 @@ const { Route } = Ember;
 export default Route.extend({
   titleToken() {
     return this.l10n.t('System Roles');
+  },
+  model() {
+    return this.get('store').findAll('user-permission');
+  },
+  actions: {
+    willTransition() {
+      this.get('controller.model').forEach(function(userPermission) {
```
Comment (niranjan94) [inline app/routes/admin/permissions/system-roles.js]: use lexical binding for uniformity.

Comment (CosmicCoder96) [inline app/routes/admin/permissions/system-roles.js]: @niranjan94 um.. do you mean should  I use something of the  sort 
`var that = this`?

Comment (niranjan94) [inline app/routes/admin/permissions/system-roles.js]: `this.get('controller.model').forEach(userPermission => {`

Comment (niranjan94) [inline app/routes/admin/permissions/system-roles.js]: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Comment by CosmicCoder96** [comment]
@niranjan94 Thanks for the reference, done.

**Pull request merged**